src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Workspace/Mef/MefWorkspaceServices.cs

```diff
@@ -51,7 +52,18 @@ public override HostServices HostServices
 
         internal IMefHostExportProvider HostExportProvider => _exportProvider;
 
-        public override Workspace Workspace => _workspace;
+        internal string WorkspaceKind => _workspace.Kind;
+
+        public override Workspace Workspace
+        {
+            get
+            {
+//#if !CODE_STYLE
+//                Contract.ThrowIfTrue(_workspace.Kind == CodeAnalysis.WorkspaceKind.RemoteWorkspace, "Access .Workspace off of a RemoteWorkspace MefWorkspaceServices is not supported.");
+//#endif
```

Comment:
can't go to this level yet.  far too many workspace services immediately grab the workspace on construction.  will need to work through avoiding that.